### PR TITLE
Fixes issue for saving model files with missing keys

### PR DESCRIPTION
**Pull request by avdudchenko**
## Fixes/Resolves:
This will catch when a result is returned with missing or inactive variables. This can occur when a solve fails or the initialization process fails without re-activating the variable first. Or where some variables will be deactivated by user based on solved condition.  

(replace this with the issue # fixed or resolved, if no issue exists then a brief statement of what this PR does)

## Summary/Motivation:
When working with NN's I build the NNs on model tree, but deactivate them. they only end up being reactivated if solve is successful, so if it fails, we will return a result with missing keys. This update will attempt to catch missing keys and rebuild them. 

Might not be the best implementation. 

needs tests

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by k1nshuk** [comment]
The changes look good to me so far. I can approve it once its out of draft.

**Comment by adam-a-a** [comment]
Based on the codecov check, it looks like you just need to add some testing.

**Comment by avdudchenko** [comment]
Looks like the tests now pass and are MPI safe. The test do need min 2 parallel workers to properly test the catches. Fails MacOS test as the objective results are not quiet right (1e-3 precision by looks of it...) 

Should be good to go/merge. This also includes some additional fixes that were caught during testing/development.

File: watertap/tools/parameter_sweep/tests/test_parameter_sweep.py
Comment (k1nshuk) [inline watertap/tools/parameter_sweep/tests/test_parameter_sweep.py]:
```suggestion
```

File: watertap/tools/parameter_sweep/tests/test_parameter_sweep.py
Comment (k1nshuk) [inline watertap/tools/parameter_sweep/tests/test_parameter_sweep.py:972]: What would be needed to make this test work with an arbitrary number of workers? This could be dealt with in a subsequent PR.

**Review by k1nshuk — APPROVED** [review]
Looks good to me so far. I am not sure about how to deal with the MacOS test, however.

File: watertap/tools/parameter_sweep/tests/test_parameter_sweep.py
Comment (avdudchenko) [reply watertap/tools/parameter_sweep/tests/test_parameter_sweep.py:972]: Its a minum number - if you have 1 worker then we don't test for missing keys. I think we should really just change how we store results from each solve (e.g instead of storing all results in arrays inside a result dictionory, we just store the value for each parameter in a dict and store an array of dicts--not sure how MPI safe that is but that would solve this issue. - this would be same as async multiprocessing or ray implementation, where we split parmans into size of 1...

**Comment by ksbeattie** [comment]
@avdudchenko still for Dec?

**Comment by avdudchenko** [comment]
> @avdudchenko still for Dec?

Would be nice, just needs one more review (@bknueven  plz) and otherwise its good to go?

**Comment by ksbeattie** [comment]
> > @avdudchenko still for Dec?
> 
> Would be nice, just needs one more review (@bknueven plz) and otherwise its good to go?

There's still a failing test :(

**Review by bknueven — APPROVED** [review]

**Pull request merged**
